keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/persistence/KeelReadOnlyRepository.kt

```diff
@@ -30,9 +31,9 @@ interface KeelReadOnlyRepository {
 
   fun constraintStateFor(deliveryConfigName: String, environmentName: String, artifactVersion: String): List<ConstraintState>
 
-  fun pendingConstraintVersionsFor(deliveryConfigName: String, environmentName: String): List<String>
+  fun getPendingArtifactVersions(deliveryConfigName: String, environmentName: String, artifact: DeliveryArtifact): List<PublishedArtifact>
```

Comment:
Renamed for clarity and consistency.

Reply:
I agree. I can do that in a separate PR. The only reason I didn't do it here has to do with something that @gal-yardeni brought up, which is the fact that these objects have a `version` field, so following the default naming pattern you end up with things like `artifactVersion.version`, which looks a bit odd. But I couldn't think of a better alternative. 🤷‍♂️